File: deploy/backup-production.pl
#!/usr/bin/env perl

use strict;
use warnings;

=head1 SYNOPSIS

  backup-production.pl

=cut

use Getopt::Long 2.33;
use Pod::Usage;

use IPC::System::Simple;
use autodie qw(:all);

sub main {
    GetOptions() or pod2usage(2);

    pod2usage(2) unless @ARGV == 0;
    my $name = 'questhub.io';

    system(qq{ssh ubuntu\@$name "sh -c 'rm -rf dump && rm -f backup.tar.gz && mongodump -d play && tar cfvz backup.tar.gz dump'"});
    system(qq{scp ubuntu\@$name:backup.tar.gz .});
    system(qq{mv backup.tar.gz ~/Dropbox/backup/$name/\$(date "+%Y-%m-%d").tar.gz});
}

main unless caller;

File: deploy/backup-production.py
#!/usr/bin/env python3

import argparse
import subprocess

def main():
    parser = argparse.ArgumentParser()
    parser.add_argument('host', nargs='?', default='questhub.io')

    args = parser.parse_args()
    host = args.host

    subprocess.check_call([
        'ssh', 'ubuntu@{}'.format(host),
        "sh -c 'rm -rf dump && rm -f backup.tar.gz && mongodump -d play && tar cfvz backup.tar.gz dump'"
    ])
    subprocess.check_call([
        'scp',
        'ubuntu@{}:backup.tar.gz'.format(host),
        '.'
    ])
    subprocess.check_call('mv backup.tar.gz ~/Dropbox/backup/' + host + '/$(date "+%Y-%m-%d").tar.gz', shell=True)

if __name__ == '__main__':
    main()
